src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+
+    private final int topSize;
+    private final int bottomSize;
+    private final int sideSize;
+
+    public IsoscelesTrapezoid(int topSize, int bottomSize, int sideSize, Colors color) {
+        super(Math.sqrt(Math.pow(topSize + bottomSize, 2)
+                * (topSize - bottomSize + 2 * sideSize)
+                * (bottomSize - topSize + 2 * sideSize)) / 4, color);
```

Comment:
For the sake of readability, it would be better to move the area calculation logic into a separate method.